### ISSUE TITLE
Voice Plugin for circuit

### ISSUE THREAD

**Issue by SaravananOffl** [circuit/circuit-sdk#46]
I'm developing a bot in Circuit which currently is doing text to text messaging, but now I need to enhance it to support voice queries also. So I've got couple of questions, How to get voice message (like voice recordings) from the user? Is there any plugin in Circuit for that?

**Comment by rogeru** [comment]
Not sure what you are trying to do, but the Circuit JS SDK allows making and receiving calls.
If you are trying to get a voicemail left for you via the SDK, then check out this jsbin.
https://output.jsbin.com/kohofiw

**Comment by SaravananOffl** [comment]
Hi, I'm actually not looking for voicemail. I want to know if there is any circuit plugin that lets the user record their voice through the mic? I want to develop a voice-based bot for circuit

**Comment by rogeru** [comment]
I am not sure what you are trying to achieve. There is no voice plugin in Circuit. Circuit uses WebRTC  for calls. No plugins are required. Using the Circuit SDK you can also make WebRTC calls.

If you want to create a bot that records a phone call, then check out https://github.com/circuit/webrtc-bot-example.

If you just want an app to record voice, then Circuit is not needed for that. Just use the getUserMedia API. See https://developers.google.com/web/fundamentals/media/recording-audio/

If this is not what you want, please explain what exactly you are trying to build.

**Comment by SaravananOffl** [comment]
I'll try to explain my project.

I'm trying to build an google assistant sort of a bot in circuit where the user can **text or  give voice commands** like  "I need document x " to the bot. The NLP Processing for the queries is done using the AWS Lex service. Now, I've achieved the **text version** of the bot through the NodeJS SDK APIs.
 
Next part of development is that I have to get voice inputs from users, and have to send it to AWS Polly for voice to text conversion. So, for this feature I'm wondering if there is some possibility to get an voice input option, like we have in Google Assistant.

Basically, I want that mic option in circuit app.
![image](https://user-images.githubusercontent.com/30234192/60167512-de738980-9820-11e9-9944-8da0e9518031.png) 

_I'm using Circuit platform for my bot because I can get user details through APIs and don't have to explicitly ask with the bot._

**Comment by rogeru** [comment]
Thanks for the explanation. Sorry there is no such option is the current Circuit clients to use voice, rather than text to send a message.

Somewhat related, there is a google assistant app for Circuit on github. Using that you can ask the assistant to send a message. It's using Google NLP and AI (Dialog Flow), not AWS. The app authenticates as yourself using OAuth, so you can get Circuit API access using the app.

For example you can say:
"Hey Google talk to Unify Circuit", then "Send `Hello World' to Allison Miller".

See https://github.com/circuit/circuit-google-assistant
